### PR TITLE
Improve formatting for GraphQLErrors

### PR DESCRIPTION
**Pull request by scratchmex**
## Description
We change the logging format at `process_errors` for `GraphQLError` errors. We use their formatting system by invoking `str()` on the class but still using the original exception via the `exc_info` parameter of the `logging` module.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* #996 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)


Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Ivan Gonzalez for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)


Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Ivan Gonzalez for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

File: strawberry/schema/schema.py [hunk 1]
```diff
@@ -177,11 +178,7 @@ def execute_sync(
         if result.errors:
             self.process_errors(result.errors, execution_context=execution_context)
 
-        return ExecutionResult(
-            data=result.data,
-            errors=result.errors,
-            extensions=result.extensions,
-        )
+        return result
```
Comment (patrick91) [inline strawberry/schema/schema.py:181]: the type of `result` is not Strawberry's `ExecutionResult`, I think we should revert this change 😊

File: tests/schema/test_execution.py [hunk 2]
```diff
@@ -174,11 +175,13 @@ def example(self) -> int:
 
     schema = strawberry.Schema(query=Query)
 
-    query = """
+    query = dedent(
+        """\
         query {
             example
-        }
+        }\
```
Comment (patrick91) [inline tests/schema/test_execution.py:182]: you could use `.strip` to skip the `\` 😊

File: strawberry/schema/schema.py [hunk 3]
```diff
@@ -103,8 +103,9 @@ def process_errors(
         self, errors: List[GraphQLError], execution_context: ExecutionContext
     ) -> None:
         for error in errors:
-            actual_error = error.original_error or error
-            logger.error(actual_error, exc_info=actual_error)
+            logger.error(
+                error, exc_info=error.original_error, stack_info=True, stacklevel=3
+            )
```
Comment (patrick91) [inline strawberry/schema/schema.py:108]: well spotted! 

@jkimbo what do you think of this change?

File: tests/schema/test_execution.py (hunk 2 above)
Comment (scratchmex) [reply tests/schema/test_execution.py:182]: Nice catch! More pythonic

Comment (scratchmex) [reply tests/schema/test_execution.py:182]: Actually, this makes `dedent` to not work as expected because it strips all the leading `\n` and we only want 1 to be removed so dedent takes out evenly those break lines:

```text
>>> print(dedent("""
...     test
...
...     GraphQL request:2:5
...     1 | query {
...     2 |     example
...       |     ^
...     3 | }
... """.strip()
... ))
test

    GraphQL request:2:5
    1 | query {
    2 |     example
      |     ^
    3 | }
```

```text
>>> print(dedent("""\
...     test
...
...     GraphQL request:2:5
...     1 | query {
...     2 |     example
...       |     ^
...     3 | }\
... """
... ))
test

GraphQL request:2:5
1 | query {
2 |     example
  |     ^
3 | }
```

What am I missing?

File: strawberry/schema/schema.py (hunk 1 above)
Comment (scratchmex) [reply strawberry/schema/schema.py:181]: image

https://github.com/strawberry-graphql/strawberry/blob/803cad533ed1a265117a3d3800265cba549e81c1/strawberry/schema/execute.py#L107

At the beginning they rename `ExecutionResult` from `graphql` to `GraphQLExecutionResult`
https://github.com/strawberry-graphql/strawberry/blob/803cad533ed1a265117a3d3800265cba549e81c1/strawberry/schema/execute.py#L7

File: strawberry/schema/schema.py (hunk 3 above)
Comment (jkimbo) [reply strawberry/schema/schema.py:108]: I had no idea you could do this. Very nice @scratchmex !

File: strawberry/schema/schema.py (hunk 1 above)
Comment (jkimbo) [reply strawberry/schema/schema.py:181]: I think @scratchmex is right @patrick91 . We're already returning the same type from `schema.execute.execute_sync`. @scratchmex can you change the return value of the `execute` method as well please?

File: tests/schema/test_execution.py (hunk 2 above)
Comment (jkimbo) [reply tests/schema/test_execution.py:182]: @scratchmex I think it should work if you put the `.strip()` on the result of `dedent`:

```python
>>> print(dedent("""
...     test
...
...     GraphQL request:2:5
...     1 | query {
...     2 |     example
...       |     ^
...     3 | }
... """
... ).strip())
```

File: strawberry/schema/schema.py (hunk 1 above)
Comment (patrick91) [reply strawberry/schema/schema.py:181]: oops, my bad!

**Comment by scratchmex** [comment]
I forced pushed for the cleanliness of the commit history. All your comments are integrated. Thanks!

File: RELEASE.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+Release type: patch
+
+We change the logging format at `process_errors` for `GraphQLError` errors.
```
Comment (patrick91) [inline RELEASE.md:3]:
``````suggestion
This release improves the default logging format for errors to include more information about the errors. For example it will show were an error was originated in a request:

```
GraphQL request:2:5
1 | query {
2 |     example
  |     ^
3 | }
```
``````

**Review by patrick91 — APPROVED** [review]
Looks good to me, I'd just change the release notes a bit 😊

**Comment by patrick91** [comment]
Uh, there are some failures here https://github.com/strawberry-graphql/strawberry/pull/1152/checks?check_run_id=3351357307

**Comment by scratchmex** [comment]
Actually is: https://docs.python.org/3/library/logging.html
image

**Comment by patrick91** [comment]
@scratchmex I pushed a change to make the code work on older python :D going to merge this in a few minutes :)

**Comment by patrick91** [comment]
@scratchmex going to invite you the org now, feel free to request a sticker from this form: https://forms.gle/dmnfQUPoY5gZbVT67
